tsdb/engine/b1/b1.go

```diff
@@ -534,6 +534,9 @@ func (tx *Tx) Cursor(key string) tsdb.Cursor {
 	// Retrieve key bucket.
 	b := tx.Bucket([]byte(key))
 
+	tx.engine.mu.Lock()
```

Comment:
@DanielMorsing tells me this lock is also taken by write code. Perhaps this lock is taken for such a short period time we should be OK?